src/components/comments/html-pin-adapter/index.ts

```diff
@@ -0,0 +1,894 @@
+   * @param {string} id the id of the element to be wrapped
+   * @returns {HTMLElement} the new wrapper element
+   */
+  private createWrapper(element: HTMLElement, id: string): HTMLElement {
+    const container = element;
+    const wrapperId = `superviz-id-${id}`;
+
+    if (this.pinsContainer.querySelector(`#${wrapperId}`)) return;
+
+    const containerRect = container.getBoundingClientRect();
+
+    const containerWrapper = document.createElement('div');
+    containerWrapper.setAttribute('data-wrapper-id', id);
+    containerWrapper.id = wrapperId;
+
+    containerWrapper.style.position = 'fixed';
+    containerWrapper.style.top = `${containerRect.top}px`;
+    containerWrapper.style.left = `${containerRect.left}px`;
+    containerWrapper.style.width = `${containerRect.width}px`;
+    containerWrapper.style.height = `${containerRect.height}px`;
+    containerWrapper.style.pointerEvents = 'none';
+    containerWrapper.style.transform = 'translateX(0) translateY(0) scale(1)';
+    containerWrapper.style.cursor = 'default';
+
+    const pinsWrapper = document.createElement('div');
+    pinsWrapper.setAttribute('data-pins-wrapper', '');
+    pinsWrapper.style.position = 'absolute';
+    pinsWrapper.style.overflow = 'hidden';
+    pinsWrapper.style.top = '0';
+    pinsWrapper.style.left = '0';
+    pinsWrapper.style.width = '100%';
+    pinsWrapper.style.height = '100%';
+    containerWrapper.appendChild(pinsWrapper);
+
+    this.pinsContainer.appendChild(containerWrapper);
+    return containerWrapper;
+  }
+
+  /**
+   * @function temporaryPinContainer
+   * @description returns the temporary pin container
+   * @returns {HTMLDivElement} the temporary pin container
+   */
+  private get temporaryPinContainer(): HTMLDivElement {
+    return document.getElementById('temporary-pin-container') as HTMLDivElement;
+  }
+
+  // ------- callbacks -------
+  /**
+   * @function onClick
+   * @description handles the click event on the container; mainly, creates or moves the temporary pin
+   * @param {MouseEvent} event the mouse event object
+   * @returns {void}
+   */
+  private onClick = (event: MouseEvent): void => {
+    if (!this.isActive || event.target === this.pins.get('temporary-pin')) return;
+
+    const target = event.target as HTMLElement;
+    const wrapper = event.currentTarget as HTMLElement;
+
+    if (target !== wrapper && this.pins.has(target.id)) {
+      return;
+    }
+
+    const elementId = wrapper.getAttribute('data-wrapper-id');
+    const rect = wrapper.getBoundingClientRect();
+    const { x: mouseDownX, y: mouseDownY } = this.mouseDownCoordinates;
+    const x = event.clientX - rect.left;
+    const y = event.clientY - rect.top;
+
+    const originalX = mouseDownX - rect.x;
+    const originalY = mouseDownY - rect.y;
+
+    const distance = Math.hypot(x - originalX, y - originalY);
+    if (distance > 10) return;
+
+    this.onPinFixedObserver.publish({
+      x,
+      y: y - 30,
```

Comment:
Why y - 30?

Reply:
So the created pin ends up exactly at the same position of the mouse cursor. Due to the cursor pointer being in the bottom-left in the original image, to appear right, it needs to be translated a little; thus, to match this, the created pins need to be translated too